doc/read-the-docs-site/architecture/runtime-view.rst

```diff
@@ -0,0 +1,169 @@
+Then, node clients can request for blocks by the using the node's chain-sync mini-protocol.
+The chain-sync protocol essentially streams events to the client which have the following structure:
+
+.. code-block:: haskell
+
+  data ChainSyncEvent a
+    = RollForward a C.ChainTip
+    | RollBackward C.ChainPoint C.ChainTip
+    deriving (Show, Functor, Generic)
+
+where ``a`` can be anything including a block.
+
+.. note::
+
+   For an overview on the different Cardano node mini-protocols, see this `Cardano Docs page <https://docs.cardano.org/explore-cardano/cardano-network/networking-protocol/>`_. For a more in-depth explanation, read the `Ouroboros network specification <https://input-output-hk.github.io/ouroboros-network/pdfs/network-spec/network-spec.pdf>`_.
+
+Now, let's assume that we want to run multiple indexers that process inputs at the same rate, and that a subset of indexers are queryable from the a client application.
+Running multiple indexers that process inputs at the same rate are managed by a ``Coordinator``.
+The runtime view would look like:
+
+.. uml::
+
+    !include <C4/C4_Dynamic>
+    title Marconi indexing runtime view with multiple indexers
+
+    Container_Ext(datasource, "Datasource")
+    Container_Ext(dApp, "dApp")
+
+    Container_Boundary(marconiContainer, "Marconi application") {
+      Component(sourceReader, "Input source reader", "Haskell component", "Module that processes data from the external datasource and transforms it to a [ProcessedInput]")
+      Component(coordinator, "Coordinator", "Haskell component", "Combines multiple indexers so that they can process inputs at the same rate")
+      Component(indexerA, "Indexer A", "Haskell component")
+      Component(indexerB, "Indexer B", "Haskell component")
+      Component(indexerC, "Indexer C", "Haskell component")
+      Component(queryAggregateD, "Read-only indexer D", "Haskell component", "Read-only indexer which doesn't store anything, but aggregates results of multiple indexers.")
+    }
+
+    Rel(datasource, sourceReader, "sends some input data")
+    Rel(sourceReader, coordinator, "sends input")
+    Rel(coordinator, indexerA, "forwards input")
+    Rel(coordinator, indexerB, "forwards input")
+    Rel(coordinator, indexerC, "forwards input")
+    Rel(queryAggregateD, indexerA, "queries")
+    Rel(queryAggregateD, indexerB, "queries")
+
+    Rel(dApp, indexerC, "queries")
+    Rel(dApp, queryAggregateD, "queries")
+
+In this view, we show that the ``Coordinator`` can combine multiple indexers.
+The ``Coordinator``'s role is to forward the incoming inputs to each of the indexer and make sure each indexer has processed the input before moving to the next one.
+
+Then, we have the querying part.
+Each indexer actually defines two things:
+
+* how to index the given input
+* how to query the indexed information
+
+Therefore, dApps will simply interact directly with the indexers when needing to query some information.
```

Comment:
It doesn't include the need for an `MVar` here. Do we want to exclude it from the explanations to keep them simpler?

Reply:
Good point. I added a note explaining how the `MVar` integrates the query and indexing thread together.